operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/DependentResourceConfiguration.java

```diff
@@ -0,0 +1,10 @@
+package io.javaoperatorsdk.operator.api.config;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+
+public interface DependentResourceConfiguration<R, P extends HasMetadata> {
+
+  Class<? extends DependentResource<R, P>> getDependentResourceClass();
+
+  Class<R> getResourceClass();
```

Comment:
This should not be `getPrimaryResourceClass()' ? or? it's not clear for me.

Reply:
No, that's actually not the primary resource class but the dependent resource one. The naming is tricky and should probably improved indeed: `getDependentResourceClass` returns the `DependentResource` implementation class (e.g. `DeploymentDependentResource`), while `getResourceClass` returns the actual associated resource class (e.g. `Deployment`).
Note that I've been trying to be consistent in the parameter types naming for these new classes: `R` represents a resource class, while `P` represents the primary resource types when it's needed. If we wanted to be more explicit we could use more explicit parameter types (e.g. use `Primary` instead of simply `P`) but that's not a common pattern in Java code…